Gradle Workflow doesn't work on Windows

```shell
C:\src> flutter create myFirstGradleAppOnWindows
Creating project myFirstGradleAppOnWindows...
C:\src> cd .\myFirstGradleAppOnWindows\
C:\src\myFirstGradleAppOnWindows> flutter run --release
Launching lib/main.dart on Nexus 6P in release mode...
Running 'gradle wrapper'...
Failed to notify ProjectEvaluationListener.afterEvaluate(), but primary configuration failure takes precedence.
java.lang.IllegalStateException: buildToolsVersion is not specified.
        at com.google.common.base.Preconditions.checkState(Preconditions.java:173)
        at com.android.build.gradle.BasePlugin.createAndroidTasks(BasePlugin.java:645)
        at com.android.build.gradle.BasePlugin$10.call(BasePlugin.java:608)
        at com.android.build.gradle.BasePlugin$10.call(BasePlugin.java:605)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:156)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:120)
        at com.android.build.gradle.BasePlugin.lambda$createTasks$1(BasePlugin.java:603)
        at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:93)
        at org.gradle.internal.event.BroadcastDispatch$ActionInvocationHandler.dispatch(BroadcastDispatch.java:82)
        at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:44)
        at org.gradle.internal.event.BroadcastDispatch.dispatch(BroadcastDispatch.java:79)
        at org.gradle.internal.event.BroadcastDispatch.dispatch(BroadcastDispatch.java:30)
        at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
        at com.sun.proxy.$Proxy15.afterEvaluate(Unknown Source)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.notifyAfterEvaluate(LifecycleProjectEvaluator.java:67)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.evaluate(LifecycleProjectEvaluator.java:61)
        at org.gradle.api.internal.project.DefaultProject.evaluate(DefaultProject.java:573)
        at org.gradle.api.internal.project.DefaultProject.evaluate(DefaultProject.java:125)
        at org.gradle.execution.TaskPathProjectEvaluator.configureHierarchy(TaskPathProjectEvaluator.java:47)
        at org.gradle.configuration.DefaultBuildConfigurer.configure(DefaultBuildConfigurer.java:38)
        at org.gradle.initialization.DefaultGradleLauncher$2.run(DefaultGradleLauncher.java:151)
        at org.gradle.internal.Factories$1.create(Factories.java:22)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:53)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:148)
        at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:33)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:112)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:106)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:63)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:92)
        at org.gradle.launcher.exec.GradleBuildController.run(GradleBuildController.java:66)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:79)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:51)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:59)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:47)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.HintGCAfterBuild.execute(HintGCAfterBuild.java:44)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:293)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
        at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\src\myFirstGradleAppOnWindows\android\app\build.gradle' line: 15

* What went wrong:
A problem occurred evaluating project ':app'.
> java.net.URISyntaxException: Illegal character in opaque part at index 5: C:srclutter/packages/flutter_tools/gradle/flutter.gradle

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Exception from flutter run: Exception: Exception: Gradle failed: 1
package:flutter_tools/src/base/common.dart 30              throwToolExit
package:flutter_tools/src/android/gradle.dart 175          ensureGradlew.<async>
dart:async                                                 _SyncCompleter.complete
package:flutter_tools/src/base/process.dart 113            runCommandAndStreamOutput.<async>
===== asynchronous gap ===========================
dart:async                                                 _Completer.completeError
package:flutter_tools/src/android/gradle.dart 186          ensureGradlew.<async>
dart:async                                                 _SyncCompleter.complete
package:flutter_tools/src/base/process.dart 113            runCommandAndStreamOutput.<async>
===== asynchronous gap ===========================
dart:async                                                 _asyncThenWrapperHelper
package:flutter_tools/src/android/gradle.dart              ensureGradlew
package:flutter_tools/src/android/gradle.dart 204          buildGradleProject.<async>
===== asynchronous gap ===========================
dart:async                                                 Future.Future.microtask
package:flutter_tools/src/android/gradle.dart              buildGradleProject
package:flutter_tools/src/commands/build_apk.dart 612      buildAndroidWithGradle.<async>
===== asynchronous gap ===========================
dart:async                                                 Future.Future.microtask
package:flutter_tools/src/commands/build_apk.dart          buildAndroidWithGradle
package:flutter_tools/src/commands/build_apk.dart 622      buildApk.<async>
===== asynchronous gap ===========================
dart:async                                                 Future.Future.microtask
package:flutter_tools/src/commands/build_apk.dart          buildApk
package:flutter_tools/src/android/android_device.dart 287  AndroidDevice.startApp.<async>
dart:async                                                 _SyncCompleter.complete
package:flutter_tools/src/base/process.dart 113            runCommandAndStreamOutput.<async>
===== asynchronous gap ===========================
dart:async                                                 _asyncThenWrapperHelper
package:flutter_tools/src/android/android_device.dart      AndroidDevice.startApp
package:flutter_tools/src/run_cold.dart 105                ColdRunner._run.<async>
dart:async                                                 _SyncCompleter.complete
package:flutter_tools/src/resident_runner.dart 185         ResidentRunner.startEchoingDeviceLog.<async>
===== asynchronous gap ===========================
dart:async                                                 _asyncThenWrapperHelper
package:flutter_tools/src/run_cold.dart                    ColdRunner._run
package:flutter_tools/src/run_cold.dart 49                 ColdRunner.run.<fn>
package:stack_trace                                        Chain.capture
package:flutter_tools/src/run_cold.dart 48                 ColdRunner.run
package:flutter_tools/src/commands/run.dart 278            RunCommand.runCommand.<async>
```

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -2,6 +2,7 @@ import java.nio.file.Path
 import java.nio.file.Paths
 
 import com.android.builder.model.AndroidProject
+import org.apache.tools.ant.taskdefs.condition.Os
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
 import org.gradle.api.Project
@@ -71,10 +72,7 @@ class FlutterPlugin implements Plugin<Project> {
             throw new GradleException("flutter.sdk must point to the Flutter SDK directory")
         }
 
-        String flutterExecutableName = localProperties.getProperty("flutter.executable")
-        if (flutterExecutableName == null) {
-            throw new GradleException("flutter.executable must be set to the name of the flutter executable")
-        }
+        String flutterExecutableName = Os.isFamily(Os.FAMILY_WINDOWS) ? "flutter.bat" : "flutter"
         flutterExecutable = Paths.get(flutterRoot.absolutePath, "bin", flutterExecutableName).toFile();
 
         String flutterJarPath = localProperties.getProperty("flutter.jar")

--- a/packages/flutter_tools/gradle/flutter.gradle
+++ b/packages/flutter_tools/gradle/flutter.gradle
@@ -28,6 +28,7 @@ apply plugin: FlutterPlugin
 
 class FlutterPlugin implements Plugin<Project> {
     private File flutterRoot
+    private File flutterExecutable
     private String localEngine
     private Properties localProperties
 
@@ -70,6 +71,12 @@ class FlutterPlugin implements Plugin<Project> {
             throw new GradleException("flutter.sdk must point to the Flutter SDK directory")
         }
 
+        String flutterExecutableName = localProperties.getProperty("flutter.executable")
+        if (flutterExecutableName == null) {
+            throw new GradleException("flutter.executable must be set to the name of the flutter executable")
+        }
+        flutterExecutable = Paths.get(flutterRoot.absolutePath, "bin", flutterExecutableName).toFile();
+
         String flutterJarPath = localProperties.getProperty("flutter.jar")
         if (flutterJarPath != null) {
             File flutterJar = project.file(flutterJarPath)
@@ -86,7 +93,7 @@ class FlutterPlugin implements Plugin<Project> {
             File releaseFlutterJar = baseEnginePath.resolve("android-arm-release").resolve("flutter.jar").toFile()
             if (!debugFlutterJar.isFile()) {
                 project.exec {
-                    executable "${flutterRoot}/bin/flutter"
+                    executable flutterExecutable.absolutePath
                     args "precache"
                 }
                 if (!debugFlutterJar.isFile()) {
@@ -139,6 +146,7 @@ class FlutterPlugin implements Plugin<Project> {
 
             FlutterTask flutterTask = project.tasks.create("flutterBuild${variant.name.capitalize()}", FlutterTask) {
                 flutterRoot this.flutterRoot
+                flutterExecutable this.flutterExecutable
                 buildMode variant.name
                 localEngine this.localEngine
                 targetPath target
@@ -164,6 +172,7 @@ class FlutterExtension {
 
 class FlutterTask extends DefaultTask {
     File flutterRoot
+    File flutterExecutable
     String buildMode
     String localEngine
     String targetPath
@@ -200,7 +209,7 @@ class FlutterTask extends DefaultTask {
 
         if (buildMode != "debug") {
             project.exec {
-                executable "${flutterRoot}/bin/flutter"
+                executable flutterExecutable.absolutePath
                 workingDir sourceDir
                 if (localEngine != null) {
                     args "--local-engine", localEngine
@@ -214,7 +223,7 @@ class FlutterTask extends DefaultTask {
         }
 
         project.exec {
-            executable "${flutterRoot}/bin/flutter"
+            executable flutterExecutable.absolutePath
             workingDir sourceDir
             if (localEngine != null) {
                 args "--local-engine", localEngine

--- a/packages/flutter_tools/lib/src/android/gradle.dart
+++ b/packages/flutter_tools/lib/src/android/gradle.dart
@@ -179,7 +179,6 @@ Future<Null> buildGradleProject(BuildMode buildMode) async {
     localProperties.writeAsStringSync(
         'sdk.dir=${_escapePath(androidSdk.directory)}\n'
         'flutter.sdk=${_escapePath(Cache.flutterRoot)}\n'
-        'flutter.executable=${platform.isWindows ? 'flutter.bat' : 'flutter'}\n'
     );
   }
   // Update the local.properties file with the build mode.

--- a/packages/flutter_tools/lib/src/android/gradle.dart
+++ b/packages/flutter_tools/lib/src/android/gradle.dart
@@ -8,6 +8,7 @@ import '../base/common.dart';
 import '../base/file_system.dart';
 import '../base/logger.dart';
 import '../base/os.dart';
+import '../base/platform.dart';
 import '../base/process.dart';
 import '../base/utils.dart';
 import '../build_info.dart';
@@ -176,8 +177,9 @@ Future<Null> buildGradleProject(BuildMode buildMode) async {
   File localProperties = fs.file('android/local.properties');
   if (!localProperties.existsSync()) {
     localProperties.writeAsStringSync(
-        'sdk.dir=${androidSdk.directory}\n'
-        'flutter.sdk=${Cache.flutterRoot}\n'
+        'sdk.dir=${_escapePath(androidSdk.directory)}\n'
+        'flutter.sdk=${_escapePath(Cache.flutterRoot)}\n'
+        'flutter.executable=${platform.isWindows ? 'flutter.bat' : 'flutter'}\n'
     );
   }
   // Update the local.properties file with the build mode.
@@ -202,6 +204,8 @@ Future<Null> buildGradleProject(BuildMode buildMode) async {
       return buildGradleProjectV2(gradlew, buildModeName);
   }
 }
+
+String _escapePath(String path) => platform.isWindows ? path.replaceAll('\\', '\\\\') : path;
 
 Future<Null> buildGradleProjectV1(String gradlew) async {
   // Run 'gradlew build'.

